dipy/segment/bundles.py

```diff
@@ -427,31 +464,71 @@ def refine(self, model_bundle, pruned_streamlines, model_clust_thr,
         first ouput of recobundle by applying second local slr (optional),
         and second pruning. This method is useful when we are dealing with
         noisy data or when we want to extract small tracks from tractograms.
+        This time, search space is created using pruned bundle and not model
+        bundle.
 
         Parameters
         ----------
         model_bundle : Streamlines
+            model bundle streamlines used as a reference to extract similar
+            streamlines from input tractogram
         pruned_streamlines : Streamlines
+            Recognized bundle from target tractogram by RecoBundles.
         model_clust_thr : float
+            MDF distance threshold for the model bundles
         reduction_thr : float
+            Reduce search space by (mm) (default 14)
         reduction_distance : string
-            mdf or mam (default mam)
+            Reduction distance type can be mdf or mam (default mdf)
         slr : bool
             Use Streamline-based Linear Registration (SLR) locally
             (default True)
         slr_metric : BundleMinDistanceMetric
-        slr_x0 : array
+        slr_x0 : array or int or str
+            Transformation allowed. translation, rigid, similarity or scaling
+            Initial parametrization for the optimization.
+
+            If 1D array with:
+                a) 6 elements then only rigid registration is performed with
+                the 3 first elements for translation and 3 for rotation.
+                b) 7 elements also isotropic scaling is performed (similarity).
+                c) 12 elements then translation, rotation (in degrees),
+                scaling and shearing are performed (affine).
+
+                Here is an example of x0 with 12 elements:
+                ``x0=np.array([0, 10, 0, 40, 0, 0, 2., 1.5, 1, 0.1, -0.5, 0])``
+
+                This has translation (0, 10, 0), rotation (40, 0, 0) in
+                degrees, scaling (2., 1.5, 1) and shearing (0.1, -0.5, 0).
+
+            If int:
+                a) 6
+                    ``x0 = np.array([0, 0, 0, 0, 0, 0])``
+                b) 7
+                    ``x0 = np.array([0, 0, 0, 0, 0, 0, 1.])``
+                c) 12
+                    ``x0 = np.array([0, 0, 0, 0, 0, 0, 1., 1., 1, 0, 0, 0])``
+
+            If str:
+                a) "rigid"
+                    ``x0 = np.array([0, 0, 0, 0, 0, 0])``
+                b) "similarity"
+                    ``x0 = np.array([0, 0, 0, 0, 0, 0, 1.])``
+                c) "affine"
+                    ``x0 = np.array([0, 0, 0, 0, 0, 0, 1., 1., 1, 0, 0, 0])
             (default None)
         slr_bounds : array
```

Comment:
`, optional` and can you add the docstring

Reply:
Hi @skoudoro,

Do you want me to put optional in front of all the following lines?

```
         If int:
                a) 6
                    ``x0 = np.array([0, 0, 0, 0, 0, 0])``
                b) 7
                    ``x0 = np.array([0, 0, 0, 0, 0, 0, 1.])``
                c) 12
                    ``x0 = np.array([0, 0, 0, 0, 0, 0, 1., 1., 1, 0, 0, 0])``

            If str:
                a) "rigid"
                    ``x0 = np.array([0, 0, 0, 0, 0, 0])``
                b) "similarity"
                    ``x0 = np.array([0, 0, 0, 0, 0, 0, 1.])``
                c) "affine"
                    ``x0 = np.array([0, 0, 0, 0, 0, 0, 1., 1., 1, 0, 0, 0])
```

Or this is enough:

`slr_x0 : array or int or str, **optional**`

Reply:
this is enough